### PR TITLE
UI changes to support removal of remote watches

### LINKED ISSUE #1003: UI changes for "Remove all remote watches"
@ibolton336 @vconzola after our conversation, will keep this GH issue updated with latest known UI requirements to support https://github.com/konveyor/mig-controller/pull/651

---

1.  **(Non display layer)** Need to set `migplan.spec.refresh=true` when re-entering the "edit" migplan wizard for an existing plan. Wait on `migplan.spec.refresh=false` and `migplan.status.conditions[RefreshInProgress].IsAbsent()` to indicate PV discovery is done before showing user PV migration options.

---

2. ~~**(Non display layer)** Ensure that _both_ launching MigStorage edit modal _and_ clicking "Check Connection" buttons for MigStorage will set `migstorage.spec.refresh: true`. Use `observedDigest` to wait for completion.~~

We do MigStorage connection checking every 30s in mig-controller. Not needed.

---

3. ~~**(Non display layer)** Ensure that _both_ launching MigCluster edit modal _and_ clicking "Check Connection" buttons for MigCluster will set `migcluster.spec.refresh: true`. Use `observedDigest` to wait for completion.~~

We do MigStorage connection checking every 30s in mig-controller. Not needed.

---

4. Moved to https://github.com/konveyor/mig-ui/issues/1012

### PR DESCRIPTION
**Pull request by djwhatle**
Hey @ibolton336 , I have started to take a first stab at solving #1003 

- Triggering `MigPlan.Spec.Refresh` at PV discovery time in UI 
- Waiting on refreshInProgress=false before showing the PV selection screen

### CODE REVIEW

File: src/app/plan/duck/types.ts [hunk 1]
```diff
@@ -46,6 +46,7 @@ export interface IMigPlan {
     destMigTokenRef?: INameNamespaceRef;
     namespaces?: string[];
     closed?: boolean;
+    refresh?: boolean;
```
Comment (djwhatle) [inline src/app/plan/duck/types.ts:49]: Added this field to MigPlan.spec

File: src/client/resources/conversions.ts [hunk 2]
```diff
@@ -440,6 +440,11 @@ export function updateMigPlanFromValues(
   if (planValues.planClosed) {
     updatedSpec.closed = true;
   }
+  if (currentPlan.status) {
```
Comment (djwhatle) [inline src/client/resources/conversions.ts:443]: Every MigPlan update issued by the UI will now set MigPlan.spec.refresh = true. There is probably a better way to do this that I'd be happy to adopt.

**Comment by djwhatle** [comment]
@ibolton336 ready for review

**Comment by djwhatle** [comment]
@ibolton336 thoughts or concerns on this PR?

**Comment by ibolton336** [comment]
> @ibolton336 thoughts or concerns on this PR?

@djwhatle Sorry this slipped through the cracks. I will review now.

File: src/client/resources/conversions.ts [hunk 3]
```diff
@@ -440,6 +440,11 @@ export function updateMigPlanFromValues(
...
+    // if currentPlan has status, this is not 1st MigPlan reconcile.
+    // we should refresh corresponding MigStorage and MigClusters.
+    updatedSpec.refresh = true;
```
Comment (ibolton336) [inline src/client/resources/conversions.ts:446]: Well done. I can't find anything wrong with this approach. This makes the UI code much more readable.  LGTM.

**Review by ibolton336 — APPROVED** [review]

**Review by ibolton336 — APPROVED** [review]
LGTM

**Pull request merged**